github/github_test.go

```diff
@@ -135,32 +134,26 @@ func testBody(t *testing.T, r *http.Request, want string) {
 	}
 }
 
-// Helper function to test that a value is marshalled to JSON as expected.
+// Test whether the marshaling of v produces a JSON that corresponds
+// to the want string.
 func testJSONMarshal(t *testing.T, v interface{}, want string) {
-	j, err := json.Marshal(v)
-	if err != nil {
-		t.Errorf("Unable to marshal JSON for %v", v)
+	// Unmarshal the wanted JSON, to verify its correctness, and marshal it back
+	// to sort the keys.
+	u := reflect.New(reflect.TypeOf(v)).Interface()
+	if err := json.Unmarshal([]byte(want), &u); err != nil {
+		t.Errorf("Unable to unmarshal JSON for %v: %v", want, err)
 	}
+	w, _ := json.Marshal(u)
```

Comment:
Please test the returned `err` here instead of ignoring it.

Reply:
done! (however, I wonder if it's ever possible for the marshal here to fail, if the immediate unmarshal had no errors)

Reply:
Probably not... but we would like to promote good Go programming practices for all developers who take a look at this code, whether they are new to Go or not... and if developers start ignoring error codes when they think there is no possible way that it could error, that opens up a potential world of pain further down the line.

Let's say, for example, we left it as-is, but then someone in the future decided to change the code above this that generated the value of `u`, but left this line the same. Suddenly, it might be possible that `u` was malformed, but this line would not catch that problem.

In summary, it is always safest to check returned error values in Go to minimize future negative surprises.  Thanks.